torch/export/dynamic_shapes.py

```diff
@@ -38,6 +42,64 @@ def readable(name, min_, max_):
             return f"Dim('{name}', min={min_})"
         return f"Dim('{name}', min={min_}, max={max_})"
 
+    def __add__(cls, other):
```

Comment:
This is a preexisting thing, but why is a metaclass needed here?

Reply:
Oh, is it so that you can put these inside type signatures? It's pretty hard to tell from this file alone because there's very little documentation on Dim

Reply:
Yeah, that whole thing is a needless complication at this point, because we're not really doing that. `Dim` would be simpler as a regular class. In fact I would love to back it with a `sympy.expr` over a user-named symbol, no need for the lambda I think.

Reply:
So, re having the symbols be a sympy.Expr, I am kind of ambivalent. I think it is OK as an API concept, but you're still going to do another translation layer under the hood to actually get it into ShapeEnv, because at the time you're assembling Dims we don't even have a ShapeEnv yet. The main place where you might screw yourself over is when Sympy made a bad API decision (like `a == b` being structural equality) and now you can't undo it without ... redoing it all yourself.

Reply:
I'm not planning to do anything here in this PR.